### PR TITLE
Make `--no-shell` not exit early

### PR DESCRIPTION
**Pull request by pbsds**
This makes `--no-shell --post-result` with multiple PRs post more than just the first report.

### CODE REVIEW

File: nixpkgs_review/cli/pr.py [hunk 1]
```diff
@@ -78,8 +78,14 @@ def pr_command(args: argparse.Namespace) -> str:
                 warn(f"https://github.com/NixOS/nixpkgs/pull/{pr} failed to build: {e}")
         assert review is not None
 
+        all_succeeded = True
         for pr, path, attrs in contexts:
-            review.start_review(attrs, path, pr, args.post_result, args.print_result)
+            succeeded = review.start_review(attrs, path, pr, args.post_result, args.print_result)
+            if not succeeded:
+                all_succeeded = False
```
Comment (Mic92) [inline nixpkgs_review/cli/pr.py:85]:
```suggestion
            succeeded &= review.start_review(attrs, path, pr, args.post_result, args.print_result)
```

File: nixpkgs_review/cli/pr.py [hunk 2]
```diff
@@ -78,8 +78,12 @@ def pr_command(args: argparse.Namespace) -> str:
                 warn(f"https://github.com/NixOS/nixpkgs/pull/{pr} failed to build: {e}")
         assert review is not None
 
+        all_succeeded = True
         for pr, path, attrs in contexts:
-            review.start_review(attrs, path, pr, args.post_result, args.print_result)
+            all_succeeded &= review.start_review(attrs, path, pr, args.post_result, args.print_result)
```
Comment (Mic92) [inline nixpkgs_review/cli/pr.py:83]:
```suggestion
            all_succeeded &= review.start_review(
              attrs, path, pr, args.post_result, args.print_result
            )
```

File: nixpkgs_review/cli/pr.py [hunk 3]
```diff
@@ -78,8 +78,14 @@ def pr_command(args: argparse.Namespace) -> str:
                 warn(f"https://github.com/NixOS/nixpkgs/pull/{pr} failed to build: {e}")
         assert review is not None
 
+        all_succeeded = True
         for pr, path, attrs in contexts:
-            review.start_review(attrs, path, pr, args.post_result, args.print_result)
+            all_succeeded &= review.start_review(
+              attrs, path, pr, args.post_result, args.print_result
+            )
```
Comment (pbsds) [inline nixpkgs_review/cli/pr.py:85]: If you prefer less procedural logic and names, then this is the more pythonic way, which does not make use of bitwise boolean operators:

```python
all_succeeded = all(
    review.start_review(attrs, path, pr, args.post_result, args.print_result)
    for pr, path, attrs in contexts
)
```

**Comment by Mic92** [comment]
@mergify queue

**Comment by Mic92** [comment]
@mergify queue

**Pull request merged**